[Fix] TAO 7321 - Media sizer not correspond size

habemuscode opened the pull request:
**NOTE: For test all functionality needs this other PR**

https://github.com/oat-sa/tao-core-ui-fe/pull/187

**Related to**

https://oat-sa.atlassian.net/browse/TAO-7321

**Description**

- Create Item.
- Add media interaction and select video for content.
- Try to change content size Width from ~1000 to ~100.

zagovorichev reviewed (changes requested):
Could you update description of the PR, it is not clear, what have you fixed

zagovorichev commented on src/qtiCommonRenderer/renderers/interactions/MediaInteraction.js:
```diff
@@ -98,8 +98,6 @@ var render = function render(interaction) {
                     renderTo: $('.media-container', $container)
                 })
                     .on('render', function() {
-                        resize();
```
But when you change browsers window size, it still calls resize and brake all that you've done, correct?

habemuscode replied:
You are right, I push a fix

zagovorichev reviewed (changes requested):
Not sure that this fix can be applied: we have to change everything, this should be a task and UI/UX approvement and test required.

zagovorichev commented on src/qtiCommonRenderer/renderers/interactions/MediaInteraction.js:
```diff
@@ -98,13 +98,8 @@ var render = function render(interaction) {
-                            .on('resize.mediaInteraction', resize);
-
-                        $item.off('resize.gridEdit').on('resize.gridEdit', resize);
+                            .off('resize.mediaInteraction');
```
Why do you need off, if you don't have on, and why have you moved this to window object and are you sure that object is still valid and it won't change component?

habemuscode replied:
I use off for ensure that events in this part are disabled, we have a lot of triggers in different parts of code

![image](https://user-images.githubusercontent.com/34692207/94801858-f2744680-03e6-11eb-8735-37ee32223755.png)

andrey-ponamarev replied:
@habemuscode I think @zagovorichev want you to remove this part of code since it does nothing:
```suggestion
         $(window).off('resize.mediaInteraction');
```
Also, I would suggest removing `resize` function since is not in used.
Are you relying on styles to change media dimensions on window resize?

zagovorichev reviewed (changes requested):
I still don't see why do we need that fix, it brakes component

habemuscode commented:
@andrey-ponamarev Thanks, more clear, I will test with and without that event.

I will remove the resize function

About your question `Are you relying on styles to change media dimensions on window resize?`

No, media interaction is in a container and should be only resized by the user, not window

zagovorichev commented:
@habemuscode  I have some questions, probably these issues are different issues of media editor (but then we need other tickets?):
1. time on the new line
![2020-10-05_17-15-02](https://user-images.githubusercontent.com/1445911/95090941-85d2b200-072e-11eb-999e-b0e1ee9087b6.png)
2. when I put width more than width of the element - with will be changed automatically, but when I put height more than expected - width will be changed to more than expected and after that it is not changed automatically. And I found that max height is set to max width value (for my case: I can't set height more than expected width, that is weird.

zagovorichev commented:
3. With this fix we've lost backgroud of the control panel

zagovorichev commented:
and if you add element after the media: then it will be under the control tools:

![2020-10-05_17-35-53](https://user-images.githubusercontent.com/1445911/95093254-3e99f080-0731-11eb-8bbf-91f0385d1359.png)

preview:
![Uploading 2020-10-05_17-36-19.png…]()

zagovorichev commented:
and sizes still show not real size of element

zagovorichev reviewed (changes requested):
I have a lot of questions about this media manager, probably would be great to have a task to rewrite it?

habemuscode commented:
@zagovorichev About point `1`

This is normal behavior, because not enough space. We can put some scroll but require UX approve and will be another ticket.

Apart from that, I consider that resize behavior will be used in a "sense way", I don't see the point of a user that resize to that smallest level

habemuscode commented:
The other points look real issues, and I see if they are easy to fix or need another ticket. Thanks for extensive test @zagovorichev!

zagovorichev commented:
> @zagovorichev About point `1`
> 
> This is normal behavior, because not enough space. We can put some scroll but require UX approve and will be another ticket.
> 
> Apart from that, I consider that resize behavior will be used in a "sense way", I don't see the point of a user that resize to that smallest level

No, there is a new line always, not for the small size only:
![2020-10-05_18-23-09](https://user-images.githubusercontent.com/1445911/95098860-e1ee0400-0737-11eb-8623-05f84ed98e64.png)

habemuscode commented:
@zagovorichev 

1. Looks fixed when I update the environment with the latest changes
2. I don't think should be fixed: When you change the height, width change according to maintain the aspect ratio of the video. Will be weirder if we have a video with width: 300px and height: 20px.... like a text lin video? You cannot see anything.
3. Fixed in the other PR: https://github.com/oat-sa/tao-core-ui-fe/pull/187
4. Fixed in the other PR: https://github.com/oat-sa/tao-core-ui-fe/pull/187

zagovorichev commented:
@habemuscode 

What about this:
![2020-10-22_17-58-10](https://user-images.githubusercontent.com/1445911/96890418-4aefae80-1490-11eb-9035-f26f2a93bb9a.png)
?

zagovorichev commented:
TBH I don't understand why is your changes better than it was before:
1. background was black (98x27)
![2020-10-22_18-06-45](https://user-images.githubusercontent.com/1445911/96892058-ecc3cb00-1491-11eb-9e00-3a6a1758902d.png)
preview:
![2020-10-22_18-11-23](https://user-images.githubusercontent.com/1445911/96892135-05cc7c00-1492-11eb-9a2f-9405048c7b06.png)
please, note: that string not in the control bar and size is the same and background of the control is black.
**And your changes:**
![2020-10-22_18-13-40](https://user-images.githubusercontent.com/1445911/96892496-5f34ab00-1492-11eb-9438-d279c5089f57.png)
form looks broken
and preview:
![2020-10-22_18-14-34](https://user-images.githubusercontent.com/1445911/96892602-783d5c00-1492-11eb-92bb-e7455261cce3.png)
it doesn't look correct

Also, @ssipasseuth or @rautenburger, could you add commentaries / correct me, please?

habemuscode commented:
@zagovorichev 

- What browser and operative system you use? Maybe is some inconsistency, for me text appears good (I use text block to test)
- The background black was set before I start working on this, I just cover with it the controls too for uniformity

zagovorichev commented:
@habemuscode 
- macOs Chrome
- I just compared itemQti without your changes (in the package.json) on the dev branch and with your changes and I'm sure that version of browser or OS are not changing this behavior

bziondik reviewed (changes requested):
Please, check height of media-player never changes.

habemuscode commented:
Was fixed by @bziondik in other PRs
